packages/block-library/src/post-author/edit.js

```diff
@@ -1,25 +1,263 @@
+		apiFetch( { path: '/wp/v2/users/' + authorId + '?context=edit' } ).then(
+			( newAuthor ) => {
+				editPost( { author: Number( authorId ) } );
+				props.setAttributes( {
+					id: newAuthor.id,
+					name: newAuthor.name,
+					avatarSize: props.attributes.avatarSize,
+					avatarUrl:
+						newAuthor.avatar_urls[ props.attributes.avatarSize ],
+					description: newAuthor.description,
+				} );
+			}
+		);
+	};
+
+	const blockClassNames = classnames( 'wp-block-post-author', {
+		[ fontSize.class ]: fontSize.class,
+	} );
+	const blockInlineStyles = {
+		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
+	};
+
+	const { isSelected } = props;
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Author Settings' ) }>
+					<SelectControl
+						label={ __( 'Author' ) }
+						value={ id }
+						options={ authors.map( ( theAuthor ) => {
+							return {
+								value: theAuthor.id,
+								label: theAuthor.name,
+							};
+						} ) }
+						onChange={ ( authorID ) => {
+							changeAuthor( authorID );
+						} }
+					/>
+					<ToggleControl
+						label={ __( 'Show avatar' ) }
+						checked={ showAvatar }
+						onChange={ () =>
+							props.setAttributes( { showAvatar: ! showAvatar } )
+						}
+					/>
+					{ showAvatar && (
+						<SelectControl
+							label={ __( 'Avatar size' ) }
+							value={ props.attributes.avatarSize }
+							options={ avatarSizes }
+							onChange={ ( size ) => {
+								props.setAttributes( {
+									avatarSize: size,
+									avatarUrl: author.avatar_urls[ size ],
+								} );
+							} }
+						/>
+					) }
+					<ToggleControl
+						label={ __( 'Show bio' ) }
+						checked={ showBio }
+						onChange={ () =>
+							props.setAttributes( { showBio: ! showBio } )
+						}
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Text settings' ) }>
+					<FontSizePicker
+						value={ fontSize.size }
+						onChange={ setFontSize }
+					/>
+				</PanelBody>
+			</InspectorControls>
+
+			{ InspectorControlsColorPanel }
+
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( nextAlign ) => {
+						props.setAttributes( { align: nextAlign } );
+					} }
+				/>
+				<BlockColorsStyleSelector
+					TextColor={ TextColor }
+					BackgroundColor={ BackgroundColor }
+				>
+					{ ColorPanel }
+				</BlockColorsStyleSelector>
+			</BlockControls>
+
+			<TextColor>
+				<BackgroundColor>
+					<div
+						ref={ ref }
+						className={ classnames( blockClassNames, {
+							[ `has-text-align-${ align }` ]: align,
+						} ) }
+						style={ blockInlineStyles }
+					>
+						{ showAvatar && (
+							<div className="wp-block-post-author__avatar">
+								<img
+									width={ props.attributes.avatarSize }
+									src={ props.attributes.avatarUrl }
+									alt={ props.attributes.name }
+								/>
+							</div>
+						) }
+						<div className="wp-block-post-author__content">
+							{ ( ! RichText.isEmpty( byline ) ||
```

Comment:
👍

Reply:
:) Thanks for the thumbs up but what for?

Reply:
I just like how it's getting hidden when empty :-)